source/Halibut.Tests/Support/TestAttributes/FriendlyHtmlSyncAndAsyncTestCaseAttribute.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections;
+using Halibut.Tests.Support.TestCases;
+using NUnit.Framework;
+
+namespace Halibut.Tests.Support.TestAttributes
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = false)]
+    public class FriendlyHtmlSyncAndAsyncTestCasesAttribute : TestCaseSourceAttribute
+    {
+        public FriendlyHtmlSyncAndAsyncTestCasesAttribute(string html, string expected)
```

Comment:
Consider something more generic to avoid an attribute per test e.g.

```
[AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = false)]
    public class SyncAndAsyncTestCaseAttribute : TestCaseSourceAttribute
    {
        public SyncAndAsyncTestCaseAttribute(params object[] parameters) :
            base(
                typeof(SyncAndAsyncTestCase),
                nameof(SyncAndAsyncTestCase.GetTestCases),
                new object []{ parameters.ToArray() })
        {
        }

        static class SyncAndAsyncTestCase
        {
            public static IEnumerable GetTestCases(object[] parameters)
            {
                var syncParameterList = new List<object>(parameters) { SyncOrAsync.Sync };
                var asyncParameterList = new List<object>(parameters) { SyncOrAsync.Async };

                yield return syncParameterList.ToArray();
                yield return asyncParameterList.ToArray();
            }
        }
    }
```

![image](https://github.com/OctopusDeploy/Halibut/assets/86938706/136d21fd-480c-406c-ba74-2dd74656c0e6)

Reply:
Ah perfect, this is what I wanted to do but I was stuck on the idea of needing a 'test case class' that wraps all the params i.e. `ClientAndServiceTestCase`. If I don't need that kind of class, and can instead use `params object[]` along with the assumption that the sync/async field is either the first or last param, then it becomes much simpler.